src/app/qgsvectorlayerproperties.cpp

```diff
@@ -681,6 +698,16 @@ void QgsVectorLayerProperties::apply()
   QgsExpressionContextUtils::setLayerVariables( mLayer, mVariableEditor->variablesInActiveScope() );
   updateVariableEditor();
 
+  // save layer dependencies
+  QSet<QString> deps;
+  Q_FOREACH ( const QgsLayerTreeLayer* layer, mLayersDependenciesTreeGroup->findLayers() )
+  {
+    if ( layer->isVisible() )
+      deps << layer->layerId();
+  }
+  QgsProject::instance()->setLayerDependencySources( mLayer->id(), deps );
+  mLayer->emitDataChanged();
```

Comment:
Is it necessary to emit data changed signal here? I would have guessed that just configuring a dependency does not need to emit this signal - confirming the dialog will always destroy the caches even if not necessary...